packages/components/src/advanced-filters/date-filter.js

```diff
@@ -132,6 +134,34 @@ class DateFilter extends Component {
 		}
 	}
 
+	onRuleChange( newRule ) {
+		const { onFilterChange } = this.props;
+		const { rule } = this.state;
+
+		let newDateState = null;
+		let shouldResetValue = false;
+
+		if ( [ rule, newRule ].includes( 'between' ) ) {
+			newDateState = {
+				before: null,
+				beforeText: '',
+				beforeError: null,
+				after: null,
+				afterText: '',
+				afterError: null,
+			};
+
+			shouldResetValue = true;
+		}
+
+		this.setState( {
+			rule: newRule,
+			...( newDateState ? newDateState : {} ),
```

Comment:
Note to self, I could avoid this ternary operator and simply do:

```
...newDateState
```

Using the spread operator on `null` is valid, e.g.

```
{
    foo: 'bar',
    ...null
}
// {foo: "bar"}
```